### ISSUE TITLE
Make Callback.ifTrue take curried parameters.

### ISSUE THREAD

**Issue by ramnivas** [japgolly/scalajs-react#264]
Currently, one needs to write code such as:
```
Callback.ifTrue(<condition>, {
   ...
   // lines of code
   ...
})
```

If `Callback.ifTrue` were to be defied as taking curried params, we could code it to look much closer to an `if` statement:

```
Callback.ifTrue(<condition>) {
   ...
   // lines of code
   ...
}
```

**Comment by japgolly** [comment]
I'd agree immediately if the second argument could be effectful but it's a `=> Callback` so it wont; it will always be a single expression.

For example, even if in the rhs you wanted to construct a new callback and embed effects (which is usually not the case for me), then currently we have:
```scala
Callback.ifTrue(_.blah == 3, Callback {
  println("Hello")
  println("World")
})
```
vs proposed
```scala
Callback.ifTrue(_.blah == 3)(Callback {
  println("Hello")
  println("World")
})
```

Do you still think it's a good improvement? I personally find the current way a little clearer.

Btw, I should rename this to `when` and add `unless` too. (#266)

**Comment by ramnivas** [comment]
I see your point. 

If `ifTure` (or `when`) is seen as a shortform for the following, then may be the curried form maps better.
```scala
if (<condition>) {
   scope.modState(...)
} else {
   Callback.empty
}
```

During recent coding I often stumbled in matching parens and braces around `ifTrue`, especially for large body of expressions. But I can see getting used to either form.

**Comment by japgolly** [comment]
Fair enough. I don't mind either way really. If you think it's clearer curried we can just do it. With 0.11 and `when`/`unless` coming soon the timing is perfect (from a backward-compatibility perspective).

**Comment by japgolly** [comment]
Closed via e0563b2a42d3ff14ed10404840f2e9f89b20d407.
